web3.js/src/connection.ts

```diff
@@ -2162,14 +2160,26 @@ export type ConfirmedSignatureInfo = {
  */
 export type HttpHeaders = {[header: string]: string};
 
+export type AlterReturnType<TFn, TReturn> = TFn extends (
+  ...args: infer A
+) => any
+  ? (...args: A) => TReturn
+  : never;
```

Comment:
I can't really make sense of this syntax.. what is being evaluated by the `?` operator?

Reply:
Fair enough. I was hoping the names of the utility types would make the reader incurious as to how it's implemented. These types should really just be TypeScript builtins, but alas.

I'll kill this stuff and just write the types manually.